### ISSUE TITLE
Merge gradle app engine facet into the app engine facet

### ISSUE THREAD

**Issue by etanshaul** [GoogleCloudPlatform/google-cloud-intellij#2157]
Follow up to #2149 

Its a better design to merge these two facets rather than having a separate one just to indicate that the user is using gradle + app-gradle-plugin.

**Comment by etanshaul** [comment]
@patflynn, @ivanporty  after thinking about it more, I'm starting to think that this is not the best move:

The configuration of the app-gradle-plugin in the build can occur independently of the project being an App Engine standard project. for example, consider this sequence (which is actually the sequence outlined in the [docs](https://cloud.google.com/appengine/docs/standard/java/tools/gradle)):

1) the user configures the app-gradle-plugin in the project
2) the user adds the appengine-web.xml to the project
3) our plugin triggers addition of app engine standard framework support

If we merge these two facets into only the single AE standard facet, then after step 3 a Gradle sync would have to occur in order to add the info to the facet (since its on sync that we detect the plugin and get data from gradle). I think this may be confusing especially since nothing in the build system changed since step 1.

In other words, by having this separate Gradle facet, then we get all the information needed immediately at the time the build.gradle is modified by the user (i.e. adding the app-gradle-plugin).

wdyt?

**Comment by patflynn** [comment]
an option is that adding an app-gradle-plugin to a project without an
appengine-web.xml shouldn't cause any facet to exist at all, since that
project is technically not runnable or deployable to app-engine.

one conceivably flow is something very roughly like:
1) normal appengine-web.xml detection triggers the user to add framework
support
2) for any module that has the appengine facet and is using gradle sync,
check for the presence of the app-gradle-plugin and configure the facet to
use it. (not sure if there's anything that would be user visible here apart
from perhaps storing any configuration state from the plugin for the run
configs)
extra credit) for any gradle synced module with the facet notify the user
if they don't have the gradle plugin installed and include the option to
automaticaly add it.

the extra credit part presents some interesting complications. i.e. for
ultimate users we'd probably want to choose between native/gradle runners
in some elegant way. (perhaps this is scope creep)

**Comment by etanshaul** [comment]
> for any module that has the appengine facet and is using gradle sync,
check for the presence of the app-gradle-plugin and configure the facet to
use it

May have misunderstood, but the problem here is that we _don't_ trigger a gradle sync on demand (nor do I think there is a good way to do this). Rather, the implementation ties into an existing gradle sync (done manually, or automatically if the user has that option configured).

So in other words, we can't check for the presence of the app-gradle-plugin (and fetch all the gradle info like build directory, etc.) unless a gradle sync happens.

**Comment by etanshaul** [comment]
and regarding the extra credit comment: we definitely want to do something like that (I metioned it as a todo in the open PR). For ultimate we wouldn't have to do anything as IDEA syncs its native build system to gradle automatically..so we'd always use the native runner.
